Update readme (#2)
DevExpress-Examples/asp-net-mvc-change-culture-on-the-fly: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T108173)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# ASP.NET MVC - How to change the current culture at runtime

> This example demonstrates how to use DevExpress [ComboBox](https://docs.devexpress.com/AspNetMvc/8984/components/data-editors-extensions/combobox) extension to change an application culture on the fly. When a user selects an item, the extension sends a request to the server to change the culture.  Between requests the combobox value is stored in cookies.

Reviewer: ```suggestion
This example demonstrates how to use the DevExpress [ComboBox](https://docs.devexpress.com/AspNetMvc/8984/components/data-editors-extensions/combobox) extension to change an application's culture on the fly. When a user selects an item, the extension sends a request to the server to change the culture.  Between requests, the combo box value is stored in cookies.
```

## Implementation Details

### 1. Add and set up a ComboBox extension.

> Wrap the ComboBox extension into the [HTML Form](https://www.w3schools.com/html/html_forms.asp) element to submit the form and send a request to the server when the selected item is changed. 

Reviewer: ```suggestion
Wrap the ComboBox extension in the [HTML Form](https://www.w3schools.com/html/html_forms.asp) element to submit the form and send a request to the server when the selected item is changed. 
```

> Handle the extension's client events to save and restore current culture (selected item) in cookies.

Reviewer: ```suggestion
Handle the extension's client events to save and restore the current culture (selected item) in cookies.
```

```cs  
@using (Html.BeginForm(null, null, FormMethod.Post, new { id = "form" })) {  
    @Html.DevExpress().ComboBox(  
        settings => {  
            settings.Name = "ComboBox";  
            settings.Width = 180;  
            settings.SelectedIndex = 0;  
            settings.Properties.ValueType = typeof(string);  
            settings.Properties.Items.Add("German", "de-DE");  
            settings.Properties.Items.Add("English", "en-US");  
            settings.Properties.ClientSideEvents.SelectedIndexChanged = "SelectedIndexChanged";  
            settings.Properties.ClientSideEvents.Init = "Init";  
        }  
    ).GetHtml()  
}  
```  

```js
function Init(s, e) {
    s.SetValue(ASPxClientUtils.GetCookie("Culture"));
}
function SelectedIndexChanged(s) {
    ASPxClientUtils.SetCookie("Culture", s.GetValue());
    $("#form").submit();
}
```

> ### 2. Change an application culture at runtime.

Reviewer: ```suggestion
### 2. Change the application culture at runtime.
```
